src/core/internal/hash.d

```diff
@@ -84,12 +84,59 @@ size_t hashOf(T)(auto ref T val, size_t seed = 0) if (!is(T == enum) && __traits
 {
     static if(__traits(isFloating, val))
     {
-        T data = (val != val) ? T.nan : val;
-        return bytesHash(toUbyte(data), seed);
+        static if (T.mant_dig == float.mant_dig && T.sizeof == uint.sizeof)
```

Comment:
This is subtly altering the behavior. NaNs aren't just one bit pattern, they are a whole set of values. The former implementation would factor this out, and treat all NaN values identically. The revised implementation has them hashing to different values.

Different NaN values should have the same hash.

Reply:
The current `hashOf` doesn't regularize NaNs when hashing an array of floats or hashing a struct that has float members. Removing the special case here made the treatment of NaNs consistent.

>Different NaN values should have the same hash.

What is the motivation for ensuring NaNs have the same hash despite not being `==`? I am thinking here about keys in hash tables.